### PR TITLE
feat(cloud): Add log reading capabilities to SyncAttempt

### PR DESCRIPTION
**Pull request by aaronsteers**
This PR targets the following PR:
- #781

---

# feat(cloud): Add log reading capabilities to SyncAttempt

## Summary

Adds two new methods to the `SyncAttempt` class to enable log retrieval from Cloud sync attempts:

- `get_log_text_tail(num_lines: int = 1000) -> str`: Returns the last N lines of log text
- `get_full_log_text() -> str`: Returns complete log text for the attempt

Both methods use the existing Config API integration (`_fetch_attempt_info()`) and handle both structured (`LogEvents`) and unstructured (`LogRead`) log formats returned by the `/v1/attempts/get_for_job` endpoint.

## Review & Testing Checklist for Human

- [ ] **Test with real log data**: Verify the methods work with actual Cloud sync attempts and return properly formatted logs
- [ ] **Verify API response format**: Confirm the Config API actually returns logs in "events" and "logLines" fields as expected from the schema analysis
- [ ] **Test edge cases**: Try with failed attempts, attempts with no logs, very large log files, and attempts with different log formats
- [ ] **Performance testing**: Check behavior with large log files to ensure `get_full_log_text()` doesn't cause memory issues

### Notes

- Implementation follows existing PyAirbyte patterns using lazy loading via `_fetch_attempt_info()`
- Field names (`events` vs `logLines`) were verified against the Config API OpenAPI schema
- Could not test locally due to environment dependency issues (missing 'six' module in pandas chain)
- Uses the established `_make_config_api_request()` pattern with proper noqa comments

**Link to Devin run**: https://app.devin.ai/sessions/946366cdd71140c88397678ef6fb16fd  
**Requested by**: @aaronsteers

### CODE REVIEW

File: airbyte/cloud/sync_results.py [hunk 1]
```diff
@@ -182,6 +182,83 @@ def _fetch_attempt_info(self) -> dict[str, Any]:
...
+    def get_log_text_tail(self, num_lines: int = 1000) -> str:
+        """Return the last N lines of log text for this attempt.
+
+        Args:
+            num_lines: Maximum number of lines to return from the end of the logs.
+                      Defaults to 1000 lines.
+
+        Returns:
+            String containing the last N lines of log text, with lines separated by newlines.
+        """
```
Comment (aaronsteers) [inline airbyte/cloud/sync_results.py]: This doesn't add anything meaningful. Let's remove this method.
